crypto/slh_dsa/slh_adrs.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/e_os2.h>
+
+/*
+ * An Address object is used to store a blob of data that is used by hash
+ * functions. It stores information related to the type of operation, as well as
+ * information related to tree addresses and heights.
+ * SHAKE based algorithms use 32 bytes for this object, whereas SHA2 based
+ * algorithms use a compressed format of 22 bytes. For this reason there are
+ * different method tables to support the different formats.
+ * FIPS 205 Section 4.2 describes the SHAKE related functions.
+ * The compressed format is discussed in Section 11.2.
+ */
+
+#define SLH_ADRS_SIZE     32 /* size of the ADRS blob */
+#define SLH_ADRSC_SIZE    22 /* size of a compact ADRS blob */
+#define SLH_ADRS_SIZE_MAX SLH_ADRS_SIZE
+
+/* 7 Different types of addresses */
+#define SLH_ADRS_TYPE_WOTS_HASH  0
+#define SLH_ADRS_TYPE_WOTS_PK    1
+#define SLH_ADRS_TYPE_TREE       2
+#define SLH_ADRS_TYPE_FORS_TREE  3
+#define SLH_ADRS_TYPE_FORS_ROOTS 4
+#define SLH_ADRS_TYPE_WOTS_PRF   5
+#define SLH_ADRS_TYPE_FORS_PRF   6
+
+typedef uint8_t *SLH_ADRS;
```

Comment:
I'm really not a fan of using a typedef to hide a pointer ... it really makes the rest of the APIs a little "strange" from a quick glance - you don't immediate know you are passing around references ...

Reply:
+1 for this change.